doc/smart-switch/pmon/smartswitch-pmon.md

```diff
@@ -436,34 +447,93 @@ is_midplane_reachable(self):
 #### 3.1.5 ModuleBase class new APIs
 
 ##### 3.1.5.1 Need for consistent storage and access of DPU reboot cause, state and health
-1.  The smartswitch needs to know the reboot cause for DPUs. Please refer to the CLI section for the various options and their effects when executed on the switch and DPUs. 
+#### 1 Reboot Cause
+The smartswitch needs to know the reboot cause for the NPU and the DPUs.
 
-* Each DPU will update its reboot cause history in the Switch ChasissStateDB upon boot up. The recent reboot-cause can be derived from that list of reboot-causes.
-* The get_reboot_cause will return the current reboot-cause of the module.
-* For persistent storage of the DPU reboot-cause and reboot-caue-history files use the existing host storage path and mechanism.
+#### NPU Reboot Cause
+* When the NPU undergoes a reboot the existing reboot-cause workflow for a switch still remains the same for a smartswitch as well.
+* When the NPU reboots the "determine_reboot_cause" fetches the npu reboot-cause and persists the files in "/host/reboot-cause" path.
+* A maximum of ten reboot-cause history entries per dpu will be persisted just like the npu.
+* The "update_reboot_cause" function updates the stateDB with reboot-cause, time, user etc as shown in the schema.
+* The existing "show reboot-cause" and "show reboot-cause history" CLIs will be backward compatible. Please refer to the CLI section.
 
-#### Schema for REBOOT_CAUSE - switch stateDB
+#### Schema for REBOOT_CAUSE of SWITCH on switch stateDB
 ```
   Key: "REBOOT_CAUSE|2023_06_18_14_56_12"
 
   "REBOOT_CAUSE|2023_06_18_14_56_12": {
     "value": {
-      "cause": "REBOOT_CAUSE_HOST_RESET_DPU",
+      "cause": "Unknown",
       "comment": "N/A",
-      "device": "DPU5",
       "time": "2023_06_18_14_56_12",
       "user": "N/A"
     }
-  },
+  }
+
+```
+#### DPU Reboot Cause
+* The smartswitch needs to know the reboot cause for all the DPUs.
+* The NPU hardware should be capable of providing the DPU reboot-cause even when the DPUs are dead.
+* The get_reboot_cause will return the current reboot-cause of the module.
+* For persistent storage of the DPU reboot-cause and reboot-cause-history files use the existing mechanism and host storage path under "/host/reboot-cause/module/dpux".
+* The storage and retrieval of the reboot-cause of the Switch and PDUs are shown in the sequence diagram
+
+<p align="center"><img src="./images/dpu-reboot-seq.svg"></p>
+
+#### Reboot workflow
+* The switch boots up. Determines the NPU reboot cause. 
+* Processes the previously stored NPU and DPU reboot-cause files and history files.
+* A maximum of ten reboot-cause history entries per dpu will be persisted just like the npu.
+* Updates the NPU reboot-cause into the StateDB and the DPU reboot-cause into the ChassisStateDB.
```

Comment:
Do we need to differentiate between NPU and DPU configuration and store one in StateDB and anther in the ChassisStateDB? Won't it be easier to store everything in StateBD?

Reply:
#closed